### DISCUSSION TITLE
Ability to inline pages and relations

### DISCUSSION THREAD

**Discussion by clementmas** [filamentphp/filament#13009] [Ideas]
I've been using Filament for several projects now and I'm consistently facing one main issue that I wish could be solved in v4.

Could we **inline the pages and relationships** of a resource?

I'm often lost with all the nested Pages and RelationManagers folders when most files are actually empty skeletons.

The simplicity of having a single file per resource is the only thing I miss from Laravel Nova. And if that could be available with Filament it would greatly improve my experience and probably make it easier to onboard new users.

Has this idea been discussed before? What do you guys think about it?

**Comment by wychoong** [comment]
> when most files are actually empty skeletons

if you delete these files, your create/edit page will become a modal when you click on the table record

> single file per resource

with this approach, its only matter of time where one will hope to break it into separate classes for the create/edit/relation page to setup form and actions. also when you combine resource with the RM, the context is mixed up, for example, OrderResource with Models/Order, with the RM for Address, Product, Customer etc



> What do you guys think about it?

at its current structure, it is clearer and provide more flexibility
